pkg/util/test/util.go

```diff
@@ -2963,6 +2980,69 @@ func ValidateTelemetry(pxImageList map[string]string, cluster *corev1.StorageClu
 	return ValidateTelemetryV1Disabled(cluster, timeout, interval)
 }
 
+// updateTelemetryStatus NOTE: is a workaround to get Telemetry again
+// from StorageCluster after validating that PX is online and upgraded
+// because there are multiple condition for when it can be disabled/enabled after sometime
```

Comment:
Can we add more details on why exactly we are doing this and what are those conditions when telemetry can be enabled/disabled? It would make it easier for the next person to understand.

Reply:
Done